.circleci/tests/test_despike.py

```diff
@@ -6,14 +6,133 @@
+    voxel_data_mean = np.mean(voxel_data)
+    voxel_data_std = np.std(voxel_data)
+    voxel_data[2] = voxel_data_mean + 10 * voxel_data_std
+    voxel_data[3] = voxel_data_mean - 10 * voxel_data_std
+
+    # What's the min and max, i.e: the amplitude of the spikes created?
+    spiked_min = np.min(voxel_data)
+    spiked_max = np.max(voxel_data)
+
+    # Let's write this temp file out for despiking
+    file_data[2, :] = voxel_data
+    tempdir = tmp_path_factory.mktemp("test_despike_nifti")
+    spikedfile = os.path.join(tempdir, "spikedfile.nii.gz")
+    despiked_file = os.path.join(tempdir, "despikedfile.nii.gz")
+    write_ndata(
+        data_matrix=file_data,
+        mask=maskfile,
+        template=boldfile,
+        TR=0.8,
+        filename=spikedfile,
+    )
+
+    # Let's despike the image and write it out to a temp file
+    despike_nifti = pe.Node(Despike(outputtype="NIFTI_GZ", args="-NEW"), name="Despike")
+    despike_nifti.inputs.in_file = spikedfile
+    despike_nifti.inputs.out_file = despiked_file
+    res = despike_nifti.run()
+
+    assert os.path.isfile(despiked_file)
+
+    file_data = read_ndata(res.outputs.out_file, maskfile)
+    voxel_data = file_data[2, :]
+
+    # What's the min and max of the despiked file?
+    despiked_min = np.min(voxel_data)
+    despiked_max = np.max(voxel_data)
+
+    # Have the spikes been reduced?
+    # i.e: has the minimum increased after despiking, and the maximum decreased?
+    # Additionally, have the affines changed?
+    assert spiked_min < despiked_min
+    assert spiked_max > despiked_max
+
+    # Are the affines the same before and after despiking?
+    assert np.array_equal(nb.load(despiked_file).affine, nb.load(boldfile).affine)
+
+
+def test_cifti_despike(data_dir, tmp_path_factory):
+    """
+    Test Cifti despiking.
+
+    Confirm that the maximum and minimum voxel values decrease
+    after despiking.
+    """
+    boldfile = os.path.join(
+        data_dir,
+        "fmriprep/sub-colornest001/ses-1/func/"
+        "sub-colornest001_ses-1_task-rest_run-1_space-"
+        "fsLR_den-91k_bold.dtseries.nii",
+    )
+    # Let's add some noise
+    file_data = read_ndata(boldfile)
+    voxel_data = file_data[2, :]  # a voxel across time
+    voxel_data_mean = np.mean(voxel_data)
+    voxel_data_std = np.std(voxel_data)
+    voxel_data[2] = voxel_data_mean + 10 * voxel_data_std
+    voxel_data[3] = voxel_data_mean - 10 * voxel_data_std
+    # What's the maximum and minimum values of the data?
+    spiked_max = max(voxel_data)
+    spiked_min = min(voxel_data)
+    # Let's write this out
+    file_data[2, :] = voxel_data
+    tempdir = tmp_path_factory.mktemp("test_despike_cifti")
+    filename = os.path.join(tempdir, "test.dtseries.nii")
+    write_ndata(data_matrix=file_data, template=boldfile, TR=0.8, filename=filename)
+    # Let's despike the data
+    # Run the node the same way it's run in XCP
+    in_file = os.path.join(tempdir, filename)
```

Comment:
You already built the input filename with `filename`. Since `tempdir` is an absolute path, this will have no effect.